src/transformState.js

```diff
@@ -4,8 +4,32 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { extraData, keysToRemove } = action;
+
+    switch (action.type) {
```

Comment:
```suggestion
    const { 
      type, 
      extraData, 
      keysToRemove 
    } = action;

    switch (type) {
```
You can get type from destructuring too